workspace/TS100/Core/Src/power.cpp

```diff
@@ -24,6 +25,21 @@ int32_t tempToX10Watts(int32_t rawTemp) {
 
 void setTipX10Watts(int32_t mw) {
 	int32_t output = X10WattsToPWM(mw, 1);
+	if (fastPWM && output == powerPWM) {
+		fastPWM = false;
+		totalPWM = 255 + 17;
+		TIM2->ARR = totalPWM;
```

Comment:
Accessing the timer directly should be pulled out to a function in `/BSP/` as its not generic across devices/boards/microcontrollers 😢

Reply:
Guess why I added [PoC] to the beginning of the commit message? ;)

Reply:
I can try to implement this idea without layering violations and with more comments but I'd prefer to do that after your testing and evaluation for obvious reasons.

Reply:
With current "slow" PWM the power can be pumped up to ~93.8 % of time, with "fast" that's lowered to ~88.2 %. I'm not sure if those ~5.5 % are worth the complexity added. OTOH the code is minimal, so probably not much harm. What I wonder is whether the "fast" mode really fast enough for all reasonable usecases.